server/plugins/IOUGens.cpp

```diff
@@ -243,11 +263,21 @@ void AudioControl_next_1(AudioControl *unit, int inNumSamples)
 			for(int i = 0; i < inNumSamples; i++){
 				out[i] = curVal;
 				curVal += valSlope;
-				}
+			}
 			unit->prevVal[0] = curVal;
 		} break;
-		case 2 :
-		    Copy(inNumSamples, out, *mapin);
+        case 2 : {
+            int thisChannelOffset = channelOffsets[unit->mSpecialIndex];
```

Comment:
this block seems like a good place for a comment (also line 221 above). maybe consider also assigning `touched[thisChannelOffset]` to a temp variable so that the `if` below is easier to read.